src/idl2json_cli/src/lib.rs

```diff
@@ -53,34 +107,17 @@ fn convert_all(
     Ok(json_structures?.join("\n"))
 }
 
-/// Get the IDL type, if specified
-fn get_idl_type(args: &Args) -> anyhow::Result<Option<IDLType>> {
-    if let (Some(did), Some(typ)) = (&args.did, &args.typ) {
-        let idl_type: IDLType = {
-            let prog = {
-                let did_as_str = std::fs::read_to_string(did)
-                    .with_context(|| anyhow!("Could not read did file '{}'.", did.display()))?;
-                IDLProg::from_str(&did_as_str)
-                    .with_context(|| anyhow!("Failed to parse did file '{}'", did.display()))?
-            };
-            polyfill::idl_prog::get(&prog, typ).ok_or_else(|| {
-                anyhow!("Type '{typ}' not found in .did file '{}'.", did.display())
-            })?
-        };
-        Ok(Some(idl_type))
-    } else {
-        Ok(None)
-    }
-}
-
 /// Converts Candid on stdin to JSON on stdout.
 #[derive(Parser, Debug, Default)]
 #[clap(name("idl2json"), version = concat!(env!("CARGO_PKG_VERSION"), "\ncandid ", env!("CARGO_CANDID_VERSION")))]
 pub struct Args {
     /// A .did file containing type definitions
     #[clap(short, long)]
-    did: Option<PathBuf>,
+    did: Vec<PathBuf>,
```

Comment:
Type definitions may be in multiple files, hence the change to `Vec`.  We don't attempt to do anything fancy, such as specifying which file to use for each type lookup.